docs/tutorials/dropout_tutorial.md

```diff
@@ -0,0 +1,234 @@
+  language: python
+  name: python3
+---
+
+# Dropout in image classification
+
+[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/google/flax/blob/main/docs/tutorials/dropout_tutorial.ipynb)
+
+This tutorial provides an end-to-end example of a simple image classification model with a dropout layer in Flax. The [dropout](https://jmlr.org/papers/volume15/srivastava14a/srivastava14a.pdf) stochastic regularization technique randomly removes hidden and visible units in a network. The randomness in Flax's [`Dropout`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.Dropout.html#flax.linen.Dropout) layer is handled internally with [`flax.linen.Module.make_rng`](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html#flax.linen.Module.make_rng). This is covered in more detail in [🔪 Flax - The Sharp Bits 🔪 `flax.linen.Dropout` layer and randomness](https://flax.readthedocs.io/en/latest/notebooks/flax_sharp_bits.html#flax-linen-dropout-layer-and-randomness).
+
+The example below uses a lot of fundamental concepts covered in [Getting started](https://flax.readthedocs.io/en/latest/getting_started.html). If you're new to Flax, start there.
+
+## Setup
+
+- Install/upgrade Flax, which will also set up [Optax](https://optax.readthedocs.io/) (for common optimizers and loss functions), and JAX.
+- Install [TensorFlow Datasets](https://www.tensorflow.org/datasets) to load a dataset for this tutorial.
+- Import the necessary libraries.
+
+```{code-cell}
+!pip install --upgrade -q flax tensorflow_datasets 
+```
+
+```{code-cell}
+import jax
+import jax.numpy as jnp                # JAX NumPy
+
+from flax import linen as nn           # The Flax Linen API
+from flax.training import train_state  # A Flax dataclass to keep the train state
+
+import numpy as np                     # Ordinary NumPy
+import optax                           # The Optax library
+import tensorflow_datasets as tfds     # TFDS for the dataset
+```
+
+Use a JAX PRNG key and split it to get one key for parameter initialization, and another one for dropout randomness:
+
+```{code-cell}
+seed = 0
+root_key = jax.random.PRNGKey(seed=seed)
+main_key, params_key, dropout_key = jax.random.split(key=root_key, num=3)
+```
+
+Create a simple Flax model, subclassed from [Flax `Module`](https://flax.readthedocs.io/en/latest/guides/flax_basics.html#module-basics). Note that:
+
+- To add a dropout layer in Flax, use [`flax.linen.Dropout`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.Dropout.html#flax.linen.Dropout).
+- In `flax.linen.Dropout`, the `deterministic` argument is `None` by default. If `false`, the inputs are scaled by `1 / (1 - dropout_rate)` and masked. When it's `true`, no mask is applied (the dropout is turned off).
+
+```{code-cell}
+# A simple convolutional network with a dropout layer.
+class CNN(nn.Module):
+  training: bool
```

Comment:
`training`  (or `train` as its most commonly used) should be an argument to `__call__`, else you cannot modify the training state via `TrainState.apply_fn`.

Reply:
Thanks @cgarciae 👍 

Just to confirm, do you mean?

```
  def __call__(self, x, train = True):
```

Reply:
Yes! Usually just `train: bool` with no default value to be very explicit.

Reply:
Yes! Usually just `train: bool` with no default value to be very explicit.